modules/network-observability-flowcollector-about.adoc

```diff
@@ -0,0 +1,147 @@
+// Module included in the following assemblies:
+
+// * configuring-operators.adoc
+
+:_content-type: CONCEPT
+[id="network-observability-flowcollector-about{context}"]
+
+= About the FlowCollector resource 
+To configure the Network Observability operator, you can modify the `FlowCollector` resource. Some of the notable configurations are documented in this documentation. 
+
+The following example shows a sample `FlowCollector` resource for {product-title} Network Observability operator:
+[id="network-observability-flowcollector-configuring-about-sample_{context}"]
+.Sample `FlowCollector` resource
+[source, yaml]
+----
```

Comment:
We should provide a smaller version, for instance:

```yaml
apiVersion: flows.netobserv.io/v1alpha1
kind: FlowCollector
metadata:
  name: cluster
spec:
  namespace: netobserv
  deploymentModel: DIRECT
  agent:
    type: EBPF
    ebpf:
      sampling: 50
      logLevel: info
      resources:
        requests:
          memory: 50Mi
          cpu: 100m
        limits:
          memory: 800Mi
  processor:
    logLevel: info
    resources:
      requests:
        memory: 100Mi
        cpu: 100m
      limits:
        memory: 800Mi
  loki:
    url: 'https://loki-gateway-http.netobserv.svc:8080/api/logs/v1/network/'
    statusUrl: 'https://loki-query-frontend-http.netobserv.svc:3100/'
    authToken: HOST
    tls:
      enable: false
  consolePlugin:
    register: true
    logLevel: info
    portNaming:
      enable: true
      portNames:
        "3100": loki
    quickFilters:
    - name: Applications
      filter:
        src_namespace!: 'openshift-,netobserv'
        dst_namespace!: 'openshift-,netobserv'
      default: true
    - name: Infrastructure
      filter:
        src_namespace: 'openshift-,netobserv'
        dst_namespace: 'openshift-,netobserv'
    - name: Pods network
      filter:
        src_kind: 'Pod'
        dst_kind: 'Pod'
      default: true
    - name: Services network
      filter:
        dst_kind: 'Service'
```

I removed some advanced configuration, IPFIX, Kafka ...

We could provide a second example with Kafka

wdyt @jpinsonneau @stleerh ?

Reply:
@nathan-weinberg @jotak okay good I'm glad you made the suggestion to condense this. I was also thinking this was quite long. We can add as much as we want regarding specific configurable components, whether its a whole example procedure or more context. Just let me know what additional configurations need their own standalone examples and/or context.

Reply:
@jotak @jpinsonneau @stleerh guidance/commands like @memodi has shared above is exactly the type of thing we should be having in the docs IMO, especially the latter

Reply:
@nathan-weinberg @stleerh @jotak @memodi --
I added a configuration procedure for increasing CPU memory limits, as Mehul provided the steps needed for that (thanks Mehul!). Can you point me to the FLP sampling rate configuration in the CRD that you're referencing? Then I can add that procedure as well.
 
It seems like we don't want to add any more specific Loki and Kafka configs because of variability with customer clusters. Is this also the case with Kafka Topic Partitions and Replicas?

Reply:
@skrthomas I'm not quite sure where it is in the CRD, but it can be patched with this command:
`$ oc patch flowcollector cluster --type=json -p "[{"op": "replace", "path": "/spec/agent/ebpf/sampling", "value": <new value>}] -n netobserv"`

I think we should definitely have Kafka Topic Partitions and Replicas documented even if we don't make specific recommendations. [Here](https://github.com/openshift-qe/ocp-qe-perfscale-ci/blob/netobserv-perf-tests/scripts/amq-streams/kafkaTopic.yaml) is the QE CRD for Kafka Topics (note this is specific to AMQ Streams as we only use this for QE Perf testing). We update Kafka Replica value with the following command:
`oc patch flowcollector/cluster --type=json -p "[{"op": "replace", "path": "/spec/processor/kafkaConsumerReplicas", "value": <new value>}]"`

I think we should also include information on configuring number of Kafka brokers, as this is something we've hit recently in QE PerfScale testing - @stleerh has information about that [here](https://coreos.slack.com/archives/C02939DP5L5/p1670351680792249).

Reply:
@nathan-weinberg tysm!!!!! I'll likely be in touch with questions after digesting these linked resources.

Reply:
@nathan-weinberg @memodi I totally understand that the CSV patch command is helpful but I'd say it's more something to add in a troubleshooting section. Normally users don't modify the CSV directly as far as I know.

So I'd suggest:
- to move that to the troubleshooting section
- on our side (dev team), raise the limit as a short term mitigation (as I commented here https://issues.redhat.com/browse/NETOBSERV-730), so the users aren't likely to run into that situation

Reply:
I would suggest we do both